models/articles.js

```diff
@@ -1,15 +1,36 @@
 const db = require("../db/connection");
-const { articleError, votesError, userError } = require("./customerrors");
-const articles = require("../db/data/test-data/articles");
+const {
+  articleError,
+  votesError,
+  userError,
+  articleColumns,
+  articleSortError,
+  articleOrderError,
+} = require("./customerrors");
 
-exports.selectAllArticles = () => {
-  return db
-    .query(
-      "SELECT articles.*, CAST((SELECT COUNT(*) AS comment_count FROM comments WHERE comments.article_id = articles.article_id) AS INT) FROM articles ORDER BY created_at DESC"
-    )
-    .then((articles) => {
-      return articles.rows;
-    });
+exports.selectAllArticles = (
+  colName = "created_at",
+  orderType = "DESC",
+  topic
+) => {
+  if (!articleColumns.includes(colName)) return articleSortError();
+  if (!["asc", "desc"].includes(orderType.toLowerCase()))
+    return articleOrderError();
+
+  let queryStr =
+    "SELECT articles.*, CAST((SELECT COUNT(*) AS comment_count FROM comments WHERE comments.article_id = articles.article_id) AS INT) FROM articles ";
```

Comment:
I'd prefer the count being done in one query as opposed to a query nested within another query - you could use a join to achieve this BUT this is very much my own opinion so if you run with this I won't judge you